canonical-web-and-design/maas-ui
# feat(ui): correctly filter KVM NUMA VMs dropdowns by node

## Done

- Updated MachineListTable to accept 'machines' and 'selectedIDs' props, to move some of the state logic into its parent component. I think eventually it might be a good idea for MachineListTable to be disconnected from the store entirely, but maybe that's something that can wait until ModularTable is ready.
- Filter aggregate VMs dropdown by those inside the KVM, and the NUMA view by each individual node.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM details page and check that the aggregate KVM dropdown correctly shows only the machines that belong to that KVM.
- Go to the NUMA view and check that each individual VM dropdown shows only shows the machines that belong to that NUMA node. Note that the API sends fake machine data, so they will be empty unless you edit the data to match known machine system ids.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2116
